#! /bin/sh /usr/share/dpatch/dpatch-run
## 054_filename_in_warning.dpatch by Stefan Fritsch <sf@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Print the filename in which the error occured

@DPATCH@
diff -urNad trunk~/server/core.c trunk/server/core.c
--- trunk~/server/core.c	2008-02-19 21:40:16.000000000 +0100
+++ trunk/server/core.c	2008-02-23 11:02:28.490211059 +0100
@@ -1356,8 +1356,8 @@
     /* Throw a warning if we're in <Location> or <Files> */
     if (ap_check_cmd_context(cmd, NOT_IN_LOCATION | NOT_IN_FILES)) {
         ap_log_error(APLOG_MARK, APLOG_WARNING, 0, cmd->server,
-                     "Useless use of AllowOverride in line %d.",
-                     cmd->directive->line_num);
+                     "Useless use of AllowOverride in line %d of %s.",
+                     cmd->directive->line_num, cmd->directive->filename);
     }
 
     d->override = OR_NONE;
